artiq/protocols/sync_struct.py

```diff
@@ -157,6 +158,8 @@ def __init__(self, notifier_name, target_builder, notify_cb=None,
                     notify_cb(mod)
         except ConnectionError:
             pass
+        except:
+            sys.excepthook(*sys.exc_info())
```

Comment:
Could you explain why this is correct?
I'd rather not assume that there is someone reading stderr and try to hit the logging infrastructure instead.

Exceptions in `_receive_cr()` should travel up through `Subscriber.close()` and then end up in the `logger` in `MasterRead._do()`.

Reply:
I think the problem actually lies in `MasterReader._do()`. If you look at this loop:

https://github.com/m-labs/artiq/blob/1fb317778a2e12af51f2f83d35f909b52cfb9c68/artiq/frontend/artiq_influxdb_schedule.py#L180-L200

...an error raised in the Subscriber (or in this case by the target) will raise at the `wait_for` on l.190. This isn't handled, so the exception bubbles up to the next try/except, but it isn't handled here either. 

It's therefore raised as a normal exception which is lost because this is happening through `ensure_future`. Worse, the loop stops so the connection is never recovered. 

The correct fix is therefore to catch the exception here and pass it to the `logger` that's available. I'll change this PR accordingly.

Reply:
Yes. Good catch!